src/strings/ops.c

```diff
@@ -1220,9 +1220,18 @@ MVMString * MVM_string_join(MVMThreadContext *tc, MVMString *separator, MVMObjec
 
             /* Add separator if needed. */
             if (i > 0) {
+                /* If there's no separator and one piece is The Empty String we
+                 * have to be extra careful about concat stability */
+                if (sgraphs == 0 && MVM_string_graphs(tc, piece) == 0) {
+                    if (!concats_stable)
+                        ;
```

Comment:
What is this empty block for? I think it would be more clear if the if statement on L1225  checked concat_stable and the `else if loop` below become just a plain if loop